### PR TITLE
doc: matter: add standalone commissioning page

### PR DESCRIPTION
**Pull request by greg-fer**
Added a separate page about commissioning.
Moved contents from the network topologies page.
Updated with more details.
NCSDK-15209.

Signed-off-by: Grzegorz Ferenc <Grzegorz.Ferenc@nordicsemi.no>

### CODE REVIEW

File: doc/nrf/ug_matter_overview_commissioning.rst [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+Device discovery
+================
+
+The commissioner discovers devices that can be commissioned onto the network.
+
+The devices that are to be commissioned need to let the commissioner know about their existence, that is *advertise* their presence.
+This can happen using the following advertising methods:
+
+* Bluetooth® LE - This method is used especially if the node is being added to its first Matter fabric.
+* Wi-Fi Access Point - This method uses the Wi-Fi Soft-AP functionality and is used for devices that are not yet members of any network.
```
Comment (Damian-Nordic) [inline doc/nrf/ug_matter_overview_commissioning.rst:61]: This is out of scope for Matter 1.0

File: doc/nrf/ug_matter_overview_commissioning.rst [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+======================
+
+The commissioner requests the commissionee to back up its original configuration.
+The fail-safe acts as a back-up, but it also starts a timer that sets a limit for the whole commissioning process.
+The timer is disabled with the disarming of the fail-safe at the end of commissioning.
+
+Preliminary node configuration
+==============================
+
+The commissioner reads the Basic Cluster Information of the commissionee and its device type.
```
Comment (Damian-Nordic) [inline doc/nrf/ug_matter_overview_commissioning.rst:86]:
```suggestion
The commissioner reads the Basic Information Cluster attributes of the commissionee and its device type.
```
Basic Information Cluster is an exact name of the cluster.

File: doc/nrf/ug_matter_overview_commissioning.rst [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+Disarming fail-safe
+===================
+
+The commissioner requests the commissionee node to remove the stored configuration backup.
+This also stops the fail-safe timer.
+
+Message exchange
+================
+
+The commissioner and the commissionee exchange an AES-encrypted message on the operational network.
```
Comment (Damian-Nordic) [inline doc/nrf/ug_matter_overview_commissioning.rst:136]: I guess this was meant to indicate that now the devices may normally talk to each other so it's not a single message. They can now exchange a number of messages.

File: doc/nrf/ug_matter_overview_commissioning.rst [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+This is because the commissioning process can take place over a different network interface (for example Bluetooth LE) than the interface used for connection with the operational network (for example Thread or Wi-Fi).
+
+Commissioning results
+*********************
+
+At the end of the commissioning procedure, the device that has successfully joined the Matter fabric will have the following information configured:
+
+* New instance name, constructed using the fabric ID and the node ID (assigned by the commissioner)
+* New Node Operational Certificate for the Matter fabric it has joined
+* New Private Key for operation certificate
```
Comment (Damian-Nordic) [inline doc/nrf/ug_matter_overview_commissioning.rst:155]:
```suggestion
* New Private Key for NOC
```
We can use the same name a before for clarity.

**Review by Damian-Nordic — APPROVED** [review]
Looks great, I have only a few minor comments.

File: doc/nrf/ug_matter_overview_commissioning.rst (hunk 3 above)
Comment (greg-fer) [reply doc/nrf/ug_matter_overview_commissioning.rst:136]: Rephrased to "The commissioner and the commissionee start exchanging AES-encrypted messages on the operational network."

File: doc/nrf/ug_matter_overview_commissioning.rst (hunk 1 above)
Comment (greg-fer) [reply doc/nrf/ug_matter_overview_commissioning.rst:61]: Should I leave this with a note "planned" or remove this?

**Review by Damian-Nordic — APPROVED** [review]
